Rework JsonFormsStateProvider

AlexandraBuzila opened the pull request:
Fixes #1574 & Fixes #1539

AlexandraBuzila committed:
Offer config customization in Standalone Renderer #1574

sdirix reviewed (changes requested):
We should merge the config with the default one. I'll take care of that.

sdirix committed:
Rework JsonFormsStateProvider

 * Initialize with Actions.init instead of delaying it
 * Reuse the config reducer

Initializing with Actions.init avoids invalid core states as we assume
that Actions.init was called from the very beginning. Usually these
invalid core states don't have consequences however sometimes the app
crashed (see #1539).

Reusing the config reducer makes sure that the config always contains at
least the default values.

sdirix reviewed (approved):
Tested in react-seed and can confirm that `config` works and the crash is also fixed.
